partner_firstname/models/res_partner.py

```diff
@@ -0,0 +1,222 @@
+# Copyright 2013 Nicolas Bessi (Camptocamp SA)
+# Copyright 2014 Agile Business Group (<http://www.agilebg.com>)
+# Copyright 2015 Grupo ESOC (<http://www.grupoesoc.es>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+
+from odoo import api, fields, models
+
+from .. import exceptions
+
+_logger = logging.getLogger(__name__)
+
+
+class ResPartner(models.Model):
+    """Adds last name and first name; name becomes a stored function field."""
+
+    _inherit = "res.partner"
+
+    firstname = fields.Char("First name", index=True)
+    lastname = fields.Char("Last name", index=True)
+    name = fields.Char(
+        compute="_compute_name",
+        inverse="_inverse_name_after_cleaning_whitespace",
+        required=False,
+        store=True,
+        readonly=False,
+    )
+
+    @api.model
+    def create(self, vals):
+        """Add inverted names at creation if unavailable."""
+        context = dict(self.env.context)
+        name = vals.get("name", context.get("default_name"))
+
+        if name is not None:
+            # Calculate the splitted fields
+            inverted = self._get_inverse_name(
+                self._get_whitespace_cleaned_name(name),
+                vals.get("is_company", self.default_get(["is_company"])["is_company"]),
+            )
+            for key, value in inverted.items():
+                if not vals.get(key) or context.get("copy"):
+                    vals[key] = value
+
+            # Remove the combined fields
+            if "name" in vals:
+                del vals["name"]
+            if "default_name" in context:
+                del context["default_name"]
+        # pylint: disable=W8121
+        return super(ResPartner, self.with_context(context)).create(vals)
```

Comment:
self.with_context(**context)) might work instead of disabling the pylint check?

Reply:
Unfortunately, I tried that. Some test were failing after parsing the context as kwargs. Will investigate further and try to make it work.

Reply:
I did try and I have all test passing

Reply:
You are right. I was mistaken, the test_create_res_partner from Odoo base code was failing. It came right after the partner_firstname tests.

Is this going to be a problem for the build?

2021-11-17 10:44:34,834 1471449 ERROR partner_firstname_db_test odoo.addons.base.tests.test_form_create: FAIL: TestFormCreate.test_create_res_partner
Traceback (most recent call last):
  File "/opt/odoo/oca/OCB/odoo/addons/base/tests/test_form_create.py", line 16, in test_create_res_partner
    partner_form.name = 'a partner'
  File "/opt/odoo/oca/OCB/odoo/tests/common.py", line 2014, in __setattr__
    assert not self._get_modifier(field, 'readonly'), \
AssertionError: can't write on readonly field name

Reply:
To be investigated further. The testcases involve a local context param on top in the class definition, which is updated in every single method. Probably been written in the past to avoid having to parse a new context every time and decrease the amount of lines of code.

Reply:
@BT-rmartin : any feedback from your side? I can't make it work with parsing the context as kwargs.

Reply:
@BT-rmartin @bjornbillen for the record: calling with_context with keyword arguments has a different effect than calling with_context with a dictionary: the first call will extend the current context while the second overwrites the current context with the dictionary. Since the code here is building a new context dictionary with values *removed* from self.env.context, you need to use the dictionary arg rather than the keyword argument form, otherwise the removed values will still be present. 

my 0.02€ ;)